tests/e2e/e2e_test.go

```diff
@@ -0,0 +1,145 @@
+package e2e
+
+import (
+	"context"
+	"crypto"
+	"fmt"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/drbg"
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/mathrand"
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature"
+	memorySigner "github.com/oasisprotocol/oasis-core/go/common/crypto/signature/signers/memory"
+	oasisGrpc "github.com/oasisprotocol/oasis-core/go/common/grpc"
+	"github.com/oasisprotocol/oasis-core/go/common/quantity"
+	"github.com/oasisprotocol/oasis-core/go/oasis-node/cmd/common"
+	"github.com/stretchr/testify/require"
+
+	"github.com/oasisprotocol/oasis-core/go/common/entity"
+	consensus "github.com/oasisprotocol/oasis-core/go/consensus/api"
+	"github.com/oasisprotocol/oasis-core/go/consensus/api/transaction"
+
+	"github.com/oasisprotocol/oasis-core/go/oasis-node/cmd/debug/txsource/workload"
+	staking "github.com/oasisprotocol/oasis-core/go/staking/api"
+	v1 "github.com/oasisprotocol/oasis-indexer/api/v1"
+	"github.com/oasisprotocol/oasis-indexer/tests"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const (
+	fundAccountAmount = 10000000000
+	timeout           = 1 * time.Second
+)
+
+func TestIndexer(t *testing.T) {
+	if _, ok := os.LookupEnv("OASIS_INDEXER_E2E"); !ok {
+		t.Skip("skipping test since e2e tests are not enabled")
+	}
+
+	if err := common.Init(); err != nil {
+		common.EarlyLogAndExit(err)
+	}
+	tests.Init()
+
+	// Indexer should be up
+	var status v1.Status
+	err := tests.GetFrom("/", &status)
+	require.Nil(t, err)
+
+	connString := "unix:/testnet/net-runner/network/validator-0/internal.sock"
+	conn, _ := oasisGrpc.Dial(connString, grpc.WithTransportCredentials(insecure.NewCredentials()))
+
+	ctx := context.Background()
+	cnsc := consensus.NewConsensusClient(conn)
+	doc, _ := cnsc.GetGenesisDocument(ctx)
+	doc.SetChainContext()
+
+	hash := crypto.SHA512
+	seed := []byte("seeeeeeeeeeeeeeeeeeeeeeeeeeeeeed")
+	name := "transfer"
+	src, _ := drbg.New(hash, seed, nil, []byte(fmt.Sprintf("txsource workload generator v1, workload %s", name)))
+
+	rng := rand.New(mathrand.New(src)) //nolint:gosec // G404: Use of weak random number generator (math/rand instead of crypto/rand)
+
+	fundingAccount, _ := memorySigner.NewFactory().Generate(signature.SignerEntity, rng)
+
+	fmt.Println("Funding account address: ", staking.NewAddress(fundingAccount.Public()))
```

Comment:
```suggestion
	t.Logf("Funding account address: %s", staking.NewAddress(fundingAccount.Public()))
```
I'd suggest using the `*testing.T` logging interface to be more idiomatic here and below.